### DISCUSSION TITLE
Open Album Performance Slow

### DISCUSSION THREAD

**Discussion by tinohager** [LycheeOrg/Lychee#2432] [Bugs] [answered]
### Lychee version

5.3.0

### Which PHP version are you using?

PHP 8.3

### Detailed description of the problem

I have some photo albums with 700 pictures, when I open one it takes 5 seconds to open. And it doesn't get any faster when I open it again. On the Linux machine I also see 100% cpu load during this time.

### Diagnostics [REQUIRED]

Diagnostics
    -----------
        Warning: Dropbox import not working. dropbox_key is empty.
        Info: Latest version of PHP is 8.3
        Warning: Default timezone not properly set; you might experience strange results when importing photos without explicit EXIF timezone
        Warning: git (software) is not available.
        Info: Found 1 medium2x that could be generated.
              You can use `php artisan lychee:generate_thumbs medium2x 1` to generate them.
    
    System Information
    ------------------
        Lychee Version (tag):                    v5.3.0 (e8f442e) -- Up to date (28 minutes ago).
        DB Version:                              5.3.0
        
        composer install:                        --no-dev
        APP_ENV:                                 production
        APP_DEBUG:                               false
        APP_URL:                                 set
        APP_DIR:                                 default
        LOG_VIEWER_ENABLED:                      true
        LIVEWIRE_ENABLED:                        true
        
        System:                                  Linux
        PHP Version:                             8.2.18
        PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
        Timezone:                                UTC
        Max uploaded file size:                  100M
        Max post size:                           100M
        Livewire chunk size:                     12.00 MB
        Max execution time:                      3600
        MySQL Version:                           10.11.7-MariaDB-1:10.11.7+maria~ubu2204
        
        exec() Available:                        yes
        Imagick Available:                       1
        Imagick Enabled:                         1
        Imagick Version:                         1691
        GD Version:                              2.3.3
        Number of foreign key:                   12 found.

### Browser & System [REQUIRED]

Docker
2 Cores
4GB

CONTAINER ID   NAME                  CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O        PIDS
2bfa90738032   lychee                115.65%   298MiB / 3.727GiB     7.81%     72.8MB / 615MB    262MB / 3.85MB   11
dad9ab77a49f   lychee_db             1.84%     212.4MiB / 3.727GiB   5.57%     7.65MB / 59.5MB   39.6MB / 121MB   11

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by tinohager** [comment]
I have reset the system to version 4.13.0, now it is super fast again.

**Reply by ildyria** [reply]
That is infuriating.

**Reply by ildyria** [reply]
I spent so much time working on Livewire. Would annoy me to migrate again to a new front-end again in pure JS (Vue3 ?)

**Comment by tinohager** [comment]
To be honest, I don't think it's because of Livewire. I develop a lot with Vue myself. I see a heavy CPU load on the server not on the client. That's why I suspect the bottleneck on the server. But I have not looked at the code.

**Reply by ildyria** [reply]
You can easily check that by adding `LIVEWIRE_ENABLED=false` and still use the version 5.3. This will make you use the JS version of the front-end instead of the Livewire.

My intuition is that Livewire is doing heavy serialization/deserialization of data in order to build the html. That increases the work load quite a bit.

**Comment by tasiotas** [comment]
did u run DB optimization?

**Reply by ildyria** [reply]
Nah, I can confirm there is a N+1 query problem I need to track down there.
It is on top of my priority list after the composer update I am currently working on.
I haven't had much time for Lychee lately due to large amount of photography work, but this is finally done, so I can get back on track. :)

**Comment by ildyria** [comment, marked as answer]
Fixed with v6. :)

**Reply by tinohager** [reply]
huge increase thank you very much
